src/main/java/org/ice4j/attribute/MessageIntegrityAttribute.java

```diff
@@ -283,19 +291,24 @@ public void decodeAttributeBody( byte[] attributeValue,
         char msgType =
             (char) (((content[0] & 0xFF) << 8) | (content[1] & 0xFF));
 
-        if(Message.isRequestType(msgType))
+        if (Message.isRequestType(msgType))
         {
             /* attribute part of a request, use the remote key */
-            key = stunStack.getCredentialsManager().getRemoteKey(username,
-                    media);
+            key
+                = stunStack.getCredentialsManager()
+                    .getRemoteKey(username, media);
         }
-        else if(Message.isSuccessResponseType(msgType) ||
-                Message.isErrorResponseType(msgType))
+        else if (Message.isResponseType(msgType))
         {
             /* attribute part of a response, use the local key */
             key = stunStack.getCredentialsManager().getLocalKey(username);
         }
+        else // msgType is Indication
+        {
+            logger.warning("Can not encode a message of type Indication.");
+        }
 
+        Objects.requireNonNull(key, "key=null; msgType=" + msgType);
```

Comment:
since we know this will only happen in the else case, is it worth just augmenting the error message logged there instead of having this here?

Reply:
I don't know whether this only happens in the "else" case, we could get `null` from the credentials manager. Actually, the reason I'm adding these logs is to try and understand why a see an exception in the logs every now and again.